### PR TITLE
Kmeans init doc update

### PR DESCRIPTION
**Pull request by olegkkruglov**
# Description
K-Means init doc update

Changes proposed in this pull request:
- Add Usage example reference to interface description
- Add random_dense method description
- Add K-Means++ description
- Add parallel K-Means++ mentioning
- Add tag-types descriptions in API section

### CODE REVIEW

File: docs/source/api/algorithms/clustering/kmeans-init.rst [hunk 1]
```diff
@@ -86,6 +86,13 @@ Operation
       | :expr:`result.centroids.row_count == desc.cluster_count`
       | :expr:`result.centroids.column_count == input.data.column_count`
 
+
+-------------
+Usage example
```
Comment (alexandraepan) [inline docs/source/api/algorithms/clustering/kmeans-init.rst:91]:
```suggestion
Example
```

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst [hunk 2]
```diff
@@ -47,6 +47,34 @@ Computing method: *dense*
...
+
+The method chooses random :math:`k` feature vectors from the training set
+:math:`X`.
+
+.. _kmeans_init_c_math_plus_plus_dense:
+
+Computing method: *plus_plus_dense* (only on CPU)
+-------------------------------------------------
+
+The method designed as follows: the first centroid :math:`c_0` is selected randomly and :math:`C = \{ c_1 \}`. Then do
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/clustering/kmeans-init.rst:63]:
```suggestion
The method is designed as follows: the first centroid :math:`c_0` is selected randomly, and :math:`C = \{ c_1 \}`. Then do
```

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst [hunk 3]
```diff
@@ -47,6 +47,34 @@ Computing method: *dense*
...
+
+.. math::
+   c_{i+1} = \mathrm{arg}\max_{x \in X} d^2(x, C), C = C \cup \{ c_{i+1} \}
+
+until :math:`C` reaches necessary size.
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/clustering/kmeans-init.rst:68]:
```suggestion
Until :math:`C` reaches the necessary size.
```

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst (hunk 2 above)
Comment (alexandraepan) [reply docs/source/onedal/algorithms/clustering/kmeans-init.rst:63]: The "Then do" part is not clear to me. At first, it seems as if you talk about the concept of the method, about something that users should not do by themselves (The method is designed as follows).  But then there is some call to action. Do users do all the steps, including the one mentioned in the first sentence?

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst [hunk 4]
```diff
@@ -47,6 +47,34 @@ Computing method: *dense*
...
+
+.. _kmeans_init_c_math_parallel_plus_dense:
+
+Computing method: *parallel_plus_dense* (only on CPU)
+-----------------------------------------------------
+
+The method is the same as :ref:`K-Means++ <kmeans_init_c_math_plus_plus_dense>` but the data is divided into equal parts and the
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/clustering/kmeans-init.rst:75]:
```suggestion
The method is the same as :ref:`K-Means++ <kmeans_init_c_math_plus_plus_dense>,` but the data is divided into equal parts and the
```

File: docs/source/api/algorithms/clustering/kmeans-init.rst (hunk 1 above)
Comment (olegkkruglov) [reply docs/source/api/algorithms/clustering/kmeans-init.rst:91]: This is a title which is used in all other sections. Don't think it needs to be changed here

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst (hunk 2 above)
Comment (olegkkruglov) [reply docs/source/onedal/algorithms/clustering/kmeans-init.rst]: you are right, it was not written well. changed, ty

File: docs/source/onedal/algorithms/clustering/kmeans-init.rst (hunk 4 above)
Comment (olegkkruglov) [reply docs/source/onedal/algorithms/clustering/kmeans-init.rst]: done

**Review by alexandraepan — APPROVED** [review]
LGTM!

**Pull request merged**